src/gfx/checked_stack.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef A67E2D4D_B267_4EFB_9A3C_C2BF0D2FE420
+#define A67E2D4D_B267_4EFB_9A3C_C2BF0D2FE420
+
+#include <vector>
+#include <stdexcept>
+#include <spdlog/fmt/fmt.h>
+
+namespace gfx {
+template <typename T> struct CheckedStack : public std::vector<T> {
+  typedef size_t Marker;
+  using std::vector<T>::size;
+  using std::vector<T>::resize;
+
+public:
+  Marker getMarker() const { return size(); }
+
+  // Validates check balance compared to marker start
+  // throws on unmatched push/pop
+  // if caught stack will be restored to the passed marker
+  void restoreMarkerChecked(Marker marker) {
+    size_t numItemsBeforeFixup = size();
+
+    // Fixup before throw in case error is handled
+    resize(marker);
+
+    if (numItemsBeforeFixup != marker) {
+      throw std::runtime_error(fmt::format("Stack imbalance ({} elements, expected {})", numItemsBeforeFixup, marker));
```

Comment:
Meh, and we give this complexity to our users?

Reply:
Should be a debug assert only maybe? @guusw was it to catch issue when developing the feature or can it still happen at runtime?

Reply:
I'll change it to `(View (-> content...))` so don't need the checks anymore

Reply:
I renamed the shard to GFX.Region to avoid confusion with the existing GFX.View (also want to rename that one to Camera later)

Reply:
It also wraps content instead of being separate push/pop

Reply:
> I'll change it to `(View (-> content...))` so don't need the checks anymore

did that happened?

Reply:
why we need still stack checks ?

Reply:
Removed scoped checks, but kept one at the end of frame to catch implementation errors